docs/ext/scope.py

```diff
@@ -0,0 +1,43 @@
+# Inspired from https://gist.github.com/kakawait/9215487 on June 14th, 2017
+# See https://github.com/sphinx-doc/sphinx/issues/1717#issuecomment-74420054
+# and https://github.com/sphinx-doc/sphinx/issues/1717#issuecomment-74679763
+
+import os, re
+from sphinx import addnodes
+
+docs_to_remove = []
+
+def setup(app):
+    app.ignore = []
+    app.connect('builder-inited', builder_inited)
+    app.connect('env-get-outdated', env_get_outdated)
+    app.connect('doctree-read', doctree_read)
+    app.add_config_value("scope_donotinclude", [], '')
+
+def builder_inited(app):
+    for doc in app.env.found_docs:
+        first_directive = None
+        for suffix in app.env.config.source_suffix:
+            _file = app.env.srcdir + os.sep + doc + suffix
+            if os.path.isfile(_file):
+                with open(_file, 'r') as fin:
+                    first_directive = fin.readline() + fin.readline()
+                if first_directive:
+                    m = re.match(r'^\.\. meta::\s+:scope: ([a-zA-Z0-9_-]+)', first_directive)
```

Comment:
if your file start with:

```
.. meta::
    :scope: <some tag>
```
It will catch the tag and look whether it is in the list `scope_donotinclude`.